src/mudlet.cpp

```diff
@@ -2641,6 +2654,153 @@ void mudlet::slot_replay()
     // No third argument causes error messages to be sent to pHost's main console:
     loadReplay(pHost, fileName);
 }
+void mudlet::generateUniqueProfileName(QString& profile_name)
+{
+    QStringList hostList = QDir(getMudletPath(profilesPath)).entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+
+    // prepend n+1 to end of the profile name
+    if (profile_name.at(profile_name.size() - 1).isDigit()) {
+        int i = 1;
+        do {
+            profile_name = profile_name.left(profile_name.size() - 1) + QString::number(profile_name.at(profile_name.size() - 1).digitValue() + i++);
+        } while (hostList.contains(profile_name));
+    } else {
+        int i = 1;
+        QString profile_name2;
+        do {
+            profile_name2 = profile_name + QString::number(i++);
+        } while (hostList.contains(profile_name2));
+        profile_name = profile_name2;
+    }
+}
+QString mudlet::addProfile(const QString& host, const int port, const QString& login, const QString& password)
+{
+    QString profile_name(host);
+    QStringList hostList = QDir(getMudletPath(profilesPath)).entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+
+    if (hostList.contains(profile_name)) {
+        generateUniqueProfileName(profile_name);
+    }
+
+    QString profilePath = mudlet::getMudletPath(mudlet::profileMapsPath, profile_name);
+    QDir dir;
+    dir.mkpath(profilePath);
+
+    writeProfileData(profile_name, QStringLiteral("url"), host);
+    writeProfileData(profile_name, QStringLiteral("port"), QString::number(port));
+
+    if (!login.isEmpty()) {
+        writeProfileData(profile_name, QStringLiteral("login"), login);
+    }
+    if (!password.isEmpty()) {
+        writeProfileData(profile_name, QStringLiteral("password"), password);
+    }
+
+    return profile_name;
+}
+
+bool mudlet::mudletIsDefault()
+{
+#if defined(Q_OS_WIN)
+    QSettings settings("HKEY_CLASSES_ROOT/telnet/shell/open/command", QSettings::NativeFormat);
+    QString value = settings.value(".", QString()).toString();
+    return value.contains("mudlet");
+#endif
+
+#if defined(Q_OS_LINUX)
```

Comment:
This should work for `Q_OS_FREEBSD` also. https://man.freebsd.org/cgi/man.cgi?query=xdg-mime&apropos=0&sektion=1&manpath=FreeBSD+13.2-RELEASE+and+Ports&arch=default&format=html
```suggestion
#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
```